pkg/prometheus/operator.go

```diff
@@ -15,21 +15,43 @@
 package prometheus
 
 import (
+	"context"
 	"fmt"
 	"reflect"
 	"regexp"
+	"strconv"
 	"strings"
+	"time"
 
 	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/operator"
 
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
 	"github.com/pkg/errors"
+	"github.com/prometheus-operator/prometheus-operator/pkg/informers"
+	appsv1 "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
 var prometheusKeyInShardStatefulSet = regexp.MustCompile("^(.+)/prometheus-(.+)-shard-[1-9][0-9]*$")
 var prometheusKeyInStatefulSet = regexp.MustCompile("^(.+)/prometheus-(.+)$")
 
+type Config struct {
```

Comment:
This can definitely be done on a follow-up, but I have the feeling we should benefit more from having an `Operator` interface that then both agent and server would comply with. Food for thought, would like to know the opinion of others.

More functions will definitely be shared among the two operators and I don't think that a `Config` struct would be the best way to pass function arguments around

Reply:
So what’s the best option now?

Reply:
> Having a shared interface can be useful now but I don't think it will stay for long 😬

I'm not sure I fully agree since in this interface we would only have methods that are indeed common to both Prom and PromAgen (in the many deployment models), for instance, we will always want to know the Status of a CR regardless of how it is deployed. Still, this was just a possible idea, I don't want to block this PR :) 

> So what’s the best option now?

I would leave the definitive answer to @simonpasquier. Personally, I think I'm more in favour of having the arguments explicit for now, but as I said, this is a slight preference as I don't think a struct called `Config` is a good attraction in this instance.

Reply:
Yeah, sorry for sounding like a blocker here. 

What I meant with different statuses with agent and server is that with, e.g. Daemonset or Sidecars, the `PrometheusStatus` interface won't make much sense to the Agent anymore since we're gonna need to tell the deployment model, the number of replicas might not be that interesting in some cases, etc.

We can definitely worry about it only when the time comes, for sure :)

Reply:
@simonpasquier I need an eye here.